zooniverse/front-end-monorepo
# Use ERAS for all user stats in YourStats component

## Package
app-project
lib-classifier

## Linked Issue and/or Talk Post
Closes: https://github.com/zooniverse/front-end-monorepo/issues/6400
Closes: https://github.com/zooniverse/front-end-monorepo/issues/5964
Closes: https://github.com/zooniverse/front-end-monorepo/issues/5134

## Describe your changes
- Remove `activity_count` everywhere in FEM so user stats are no longer using `/project_preferences` for calculating classification count.
- Untangle the spaghetti code of UPP > YourStats MST files.
- Add a disposer function with autorun to Store.js so that UserPersonalization refreshes every time the project id changes. This ensures the YourStats component displays correct stats data if you keep the same tab open and switch between FEM projects.

## How to Review

- View any project's classify page and YourStats total classification count should match the total classification count displayed on your stats and homepage.
- When you make a classification, the YourStats component should increment as expected.
- See more comments below about removing legacy code.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)

## Bug Fix

## Refactoring
